File: JobScheduler/app/src/main/java/com/example/jobscheduler/MainActivity.java
package com.example.jobscheduler;

import androidx.appcompat.app.AppCompatActivity;

import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.ComponentName;
import android.content.Context;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private static final int JOB_ID = 123;

    //https://medium.com/google-developers/scheduling-jobs-like-a-pro-with-jobscheduler-286ef8510129
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void sjob(View view) {

        JobScheduler jobScheduler =
                (JobScheduler) getSystemService(Context.JOB_SCHEDULER_SERVICE);

        JobInfo jobInfo= new JobInfo.Builder(JOB_ID,new ComponentName(this,DemoJob.class)).setPeriodic(120000)
                .setPersisted(true).setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED).setRequiresCharging(true).build();

        jobScheduler.schedule(jobInfo);

    }
}

File: Notifications/app/src/main/java/com/example/notifications/MainActivity.kt
package com.example.notifications

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.app.NotificationCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val nm = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
            nm.createNotificationChannel(NotificationChannel("first channel",
            "default",
            NotificationManager.IMPORTANCE_DEFAULT))
        }
        btn1.setOnClickListener {
            val simpleNotification = NotificationCompat.Builder(this,"first channel")
                    .setContentTitle("SimpleTitle")
                    .setContentText("this is the sample description of the notification")
                    .setSmallIcon(R.drawable.ic_launcher_foreground)
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                    .build()
            nm.notify(1,simpleNotification)
        }
        btn2.setOnClickListener {
            val simpleNotification = NotificationCompat.Builder(this,"first channel")
                    .setContentTitle("SimpleTitle")
                    .setContentText("hi i m Yash")
                    .setSmallIcon(R.drawable.ic_launcher_foreground)
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                    .build()
            nm.notify(1,simpleNotification)
        }
        btn3.setOnClickListener {
            val simpleNotification = NotificationCompat.Builder(this,"first channel")
                    .setContentTitle("SimpleTitle")
                    .setContentText("this is a sample")
                    .setSmallIcon(R.drawable.ic_launcher_foreground)
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                    .build()
            nm.notify(1,simpleNotification)
        }
    }
}